### PR TITLE
Layouts and menu

### PR DESCRIPTION
**Pull request by jamesthebrooks**
## Changes proposed in this pull request:
- Layout files migrated over
- Main menu shows default items as data is migrated in another branch

### CODE REVIEW

File: _includes/collection-item.html
Comment (sknep) [inline _includes/collection-item.html:1]: does the name of this component still make sense if you're pulling the collection item class out? also if these are items in a list, they should still be `<li>`s

File: _includes/doc-registry.html
Comment (sknep) [inline _includes/doc-registry.html:1]: just curious, why the assignment? you can loop thru  `site.documentation_navigation` on the next line without assigning it.

File: _includes/doc-registry.html
Comment (sknep) [inline _includes/doc-registry.html:4]: these are the font awesome icon classes. Was it not possible to migrate these to USWDS icons?

File: _includes/footer.html
Comment (sknep) [inline _includes/footer.html:2]: can you tell me why you went full utility classes here instead of sticking with the identifier classes? I can't tell what's better about this implementation.

File: _includes/footer.html
Comment (sknep) [inline _includes/footer.html:12]: IMO all external (and mailto) links should open in a new window. Also, there's no way you could be familiar with this already, but here's some guidance: https://github.com/cloud-gov/product/blob/main/ContentGuide.md#2-referring-to-external-or-non-government-entities

also join.tts.gsa.gov recently moved to https://tts.gsa.gov/join/

File: _includes/footer.html
Comment (sknep) [inline _includes/footer.html:9]: would you be willing to put these footer links into a yaml/json/whatever list in the data folder and loop through them rather than hardcode the link text and hrefs into the template? I think it'd be an improvement from a content management perspective.

File: _includes/footer.html
Comment (sknep) [inline _includes/footer.html:25]: new window for this link please :)

File: _includes/footer.html
Comment (sknep) [inline _includes/footer.html:37]: it's long-since time we made our own copy of this open source policy for our own site and linked to it instead. paging @wz-gsa to make sure we're allowed to do that. I'm so tired of linking to 18Fs stuff when we've been our own program for years now.

File: _includes/layouts/base.html
Comment (sknep) [inline _includes/layouts/base.html:22]: we use assign in so many places, it seems obv to do it here too. would something like this work? note that you'd need to rename the final keys for one of them (`primary` -> `primary_navigation` or vice-vera )
```
{% assign nav = site %}
{% if "pages" %} {% nav = pages.navigation %} {% endif %}
{% include "menu.html" primary_navigation: nav.primary secondary_navigation: nav.secondary %}
  
```

**Review by sknep — CHANGES_REQUESTED** [review]
Don't be afraid to re-name or reorganize some variables to make this nav stuff less annoying to understand

File: _includes/collection-item.html
Comment (jamesthebrooks) [reply _includes/collection-item.html:1]: This change was mostly to keep the look and feel of the existing cg-site--I'll defer to you on this--pages way or pre-existing cg-site look?  They probably shouldn't be in a list as these are multi-part paginated code blocks.  Semantically, it's fine either way.

File: _includes/doc-registry.html
Comment (jamesthebrooks) [reply _includes/doc-registry.html:1]: No idea.  Not my change, but is a 1:1 from the old site.  I'm updating to remove the assignment, but not actually sure where this template is used off the top of my head.

File: _includes/doc-registry.html
Comment (jamesthebrooks) [reply _includes/doc-registry.html:4]: I don't even see a reference to this template. Deleting the file.

File: _includes/footer.html
Comment (jamesthebrooks) [reply _includes/footer.html]: I updated this to override 11ty's default URL filter and am using it {{ somelink | url }} in the template here going forward.  Not ready to merge yet.

File: _includes/layouts/docs.html
Comment (jamesthebrooks) [reply _includes/layouts/docs.html]: Simplified the logic by moving it to eleventy.js.  Compiles fine, but won't be able to feature test until both this and documentation is merged in (PR not yet created).

File: _includes/layouts/docs.html
Comment (jamesthebrooks) [reply _includes/layouts/docs.html:35]: Updated in .eleventy.js

File: _includes/layouts/page.html
Comment (jamesthebrooks) [reply _includes/layouts/page.html:9]: It comes from any given page's front matter so you'd set it there to whatever makes sense.

Example- some-page.html:
---
background: bg-primary-dark
---

It's not a convention I came up with... it's either from cg-site or pages-11ty (can't remember).

File: _includes/layouts/post.html
Comment (jamesthebrooks) [reply _includes/layouts/post.html:23]: See .eleventy.js

File: .eleventy.js
Comment (sknep) [inline .eleventy.js:48]: For every new function/filter you add to the eleventy config, do me a solid and comment what it's for. I can kinda tell but it's always nice to have a little more what/why <3

File: .eleventy.js
Comment (sknep) [inline .eleventy.js:169]: this is awesome and makes me happy. If you also add a class ("usa-link--external") I think we get the little visual indicator [docs](https://designsystem.digital.gov/components/link/#component-variants-link) which would be nice in the article content.

File: .eleventy.js
Comment (sknep) [inline .eleventy.js:200]: oh maybe this is where to put the external link class? not the above one?

File: _includes/layouts/page.html
Comment (sknep) [inline _includes/layouts/page.html:33]: can this one be in a new window too?

File: js/app.js
Comment (sknep) [inline js/app.js:52]: Can this be done more simply, with a parent class that handles the toggling? this is a lot of dom manipulation for what is effectively a few CSS changes.

**Comment by sknep** [comment]
Something's broken in this build, see the build logs:

```2024-10-31 15:23:23 INFO [run-federalist-script] [11ty]

2024-10-31 15:23:23 INFO [run-federalist-script] [11ty] Original error stack trace: expected "|" before filter, file:/tmp/work/site_repo/_includes/footer.html, line:16, col:55

2024-10-31 15:23:23 INFO [run-federalist-script] [11ty]    14|                                 </li>

2024-10-31 15:23:23 INFO [run-federalist-script] [11ty]    15|                                 <li class="usa-footer__secondary-link">

2024-10-31 15:23:23 INFO [run-federalist-script] [11ty] >> 16|                                     <a href="{{ mailto:inquiries@cloud.gov | url }}">Business inquiries</a>

2024-10-31 15:23:23 INFO [run-federalist-script] [11ty]                                                              ^

2024-10-31 15:23:23 INFO [run-federalist-script] [11ty]    17|                                 </li>

2024-10-31 15:23:23 INFO [run-federalist-script] [11ty]    18|                                 <li class="usa-footer__secondary-link">

2024-10-31 15:23:23 INFO [run-federalist-script] [11ty]    19|                                     <a href="{{ 'https://join.tts.gsa.gov/' | url }}">Join our team</a>

2024-10-31 15:23:23 ERROR [main] There was a problem running the federalist script, see the above logs for details.

```

File: _includes/layouts/post.html
Comment (jamesthebrooks) [reply _includes/layouts/post.html:23]: I think you already saw it by now because of another comment, but here: https://github.com/cloud-gov/site/pull/20/files#diff-c306e0a99961a16f5c5c83996caa0958b94006d97f97475049ea3a08036bb5b0

Eleventy now handles external links on all content and templates during the build through a custom function on the markdown-it plugin.

File: .eleventy.js
Comment (jamesthebrooks) [reply .eleventy.js:200]: Should be both.  One is for markdown and the other is for html files.

File: js/app.js
Comment (jamesthebrooks) [reply js/app.js:52]: Probably, but if I make changes here I can't test them as the sidenav doesn't exist in this branch.  We would have to wait until the PR for branch chore-dynamic-sidenav is up for review--that branch and PR does not exist yet.  This block is a 1:1 from the Jekyll site with minimal changes to make it continue to function.  Please let me know how you'd like me to proceed.

File: .eleventy.js
Comment (jamesthebrooks) [reply .eleventy.js:50]: In another comment on the docs.html page, you asked for the template markup to be simpler and so I moved the logic to here.  The new docs.html code looks like this now:

                {% assign toc_content = content | prepTOCContent %}
                {% if toc_content | shouldShowTOC: showtoc %}
                <h2>Table of Contents</h2>
                <div id="table-of-contents">
                    {{ toc_content }}
                </div>
                {% endif %}
                
The markup is simpler now, but the logic still needs to exist somewhere.

Comment (sknep) [reply .eleventy.js:50]: If you're going to add custom functions/filters (magic) to eleventy, it needs to be commented in place for other people to understand what it is and how it works. Maybe you thought I meant here on the PR?

File: _includes/layouts/post.html
Comment (sknep) [reply _includes/layouts/post.html:23]: > Eleventy now handles external links on all content and templates during the build through a custom function on the markdown-it plugin.

Hmm... That would work on markdown file content parsed by the markdown engine. This part is in a template. Are you sure it still works that way?

**Comment by sknep** [comment]
Is the build still broken? I think that mailto link might not work with your custom HTML link fixer

File: _includes/layouts/post.html
Comment (jamesthebrooks) [reply _includes/layouts/post.html:23]: 100%.  Side note, adding the class "usa-link--external" makes footer links that are external very hard to read.  Override it for the footer?  Run local to view or shoot me a message in slack and I'll send you a screenshot.

File: .eleventy.js
Comment (jamesthebrooks) [reply .eleventy.js:50]: I did both so it's tracked both in the code and on the github discussion.

**Comment by jamesthebrooks** [comment]
> Is the build still broken? I think that mailto link might not work with your custom HTML link fixer

It's good to go now.

File: .eleventy.js
Comment (jamesthebrooks) [reply .eleventy.js:203]: Okay, I'll revert this change.

File: _includes/layouts/post.html
Comment (sknep) [reply _includes/layouts/post.html:23]: ok so we still need to target _blank stuff here, hardcoded, yes?

File: _includes/layouts/page.html
Comment (sknep) [reply _includes/layouts/page.html:33]: now that we cut the jsdom stuff, this needs that attribute manually added

File: _includes/layouts/post.html
Comment (jamesthebrooks) [reply _includes/layouts/post.html:23]: I got that too.

**Review by sknep — APPROVED** [review]
onward! and thank you

**Pull request merged**
